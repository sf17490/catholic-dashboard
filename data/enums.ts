//export type DioceseName = 'Arundel & Brighton' | 'Birmingham' | 'Brentwood' | 'Cardiff-Menevia' | 'Clifton' | 'East Anglia' | 'Hallam' | 'Hexham & Newcastle' | 'Lancaster' | 'Leeds' | 'Liverpool' | 'Middlesbrough' | 'Northampton' | 'Nottingham' | 'Plymouth' | 'Portsmouth' | 'Salford' | 'Shrewsbury' | 'Southwark' | 'Westminster' | 'Wrexham'
export const validDioceseNames = [
  "Arundel & Brighton",
  "Birmingham",
  "Brentwood",
  "Cardiff-Menevia",
  "Clifton",
  "East Anglia",
  "Hallam",
  "Hexham & Newcastle",
  "Lancaster",
  "Leeds",
  "Liverpool",
  "Middlesbrough",
  "Northampton",
  "Nottingham",
  "Plymouth",
  "Portsmouth",
  "Salford",
  "Shrewsbury",
  "Southwark",
  "Westminster",
  "Wrexham",
] as const;
export type DioceseName = (typeof validDioceseNames)[number];
