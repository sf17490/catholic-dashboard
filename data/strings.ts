export const CONVERSIONS_ACCURACY_STATEMENT =
  "The data below records receptions into the Catholic Church. It does not include figures for adult baptisms.";
export const MA_ACCURACY_STATEMENT =
  "The accuracy with which attendance is counted may vary. A person may accidentally be counted twice and figures may overstate or understate where an estimate is required, such as at very large services or if mechanical means of counting fail. Additionally, approaches to counting may vary across churches in a diocese or nation.";
