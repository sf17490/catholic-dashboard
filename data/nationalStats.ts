import { DbKey } from "./enums";
import { LineGraphProps } from "@/pages/components/PlotALineGraph";

interface NationalData {
  chartData: {
    heading: string;
    contextParagraph: string;
    lineGraphData: LineGraphProps;
  };
}

const nationalConversions: LineGraphProps = {
  yAxisLabel: "Converts to Catholicism",
  yAxisValues: [4402, 4615, 2674, 1537, 1976],
  xAxisLabel: "Year",
  xAxisValues: [2000, 2010, 2019, 2021, 2022],
};

const nationalConversionsData: NationalData = {
  chartData: {
    heading: "Conversions to Catholicism in England & Wales",
    contextParagraph: "Receptions into the Catholic Church in England & Wales (Is this adults only?)",
    lineGraphData: nationalConversions,
  },
};

const nationalMassAttendance: LineGraphProps = {
  yAxisLabel: "Number attending Mass",
  yAxisValues: [1005522, 885169, 712909, 390236, 503008],
  xAxisLabel: "Year",
  xAxisValues: [1999, 2008, 2018, 2021, 2022],
};

const nationalMassAttendanceData: NationalData = {
  chartData: {
    heading: "Typical Sunday Mass Attendance",
    contextParagraph: "Typical number of people attending Sunday Mass in England & Wales",
    lineGraphData: nationalMassAttendance,
  },
};



export const SimpleDb2 = {
  nationalConversions: nationalConversionsData,
  nationalMassAttendance: nationalMassAttendanceData
}

export function getNationalData(keyName: DbKey){
  return SimpleDb2[keyName]
}

export const SimpleDb = {
  nationalMassAttendance: nationalMassAttendance,
  nationalConversions: nationalConversions,
};

export function getData(keyName: DbKey) {
  return SimpleDb[keyName];
}
