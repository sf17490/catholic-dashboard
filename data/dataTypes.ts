import { TableColumns, TableRow } from "@/components/Table";
import { DioceseSimpleDb } from "./dioceseStats";
import { SimpleDb } from "./nationalStats";
export const validDioceseNames = [
  "Arundel & Brighton",
  "Birmingham",
  "Brentwood",
  "Cardiff-Menevia",
  "Clifton",
  "East Anglia",
  "Hallam",
  "Hexham & Newcastle",
  "Lancaster",
  "Leeds",
  "Liverpool",
  "Middlesbrough",
  "Northampton",
  "Nottingham",
  "Plymouth",
  "Portsmouth",
  "Salford",
  "Shrewsbury",
  "Southwark",
  "Westminster",
  "Wrexham",
] as const;
export type DioceseName = (typeof validDioceseNames)[number];

export const cleanedDioceseNames = [
  "arundel_and_brighton",
  "birmingham",
  "brentwood",
  "cardiff-menevia",
  "clifton",
  "east_anglia",
  "hallam",
  "hexham_and_newcastle",
  "lancaster",
  "leeds",
  "liverpool",
  "middlesbrough",
  "northampton",
  "nottingham",
  "plymouth",
  "portsmouth",
  "salford",
  "shrewsbury",
  "southwark",
  "westminster",
  "wrexham",
] as const;
export type CleanedDioceseName = (typeof cleanedDioceseNames)[number];

export type DbKey = keyof typeof SimpleDb;
export type DioceseDbKey = keyof typeof DioceseSimpleDb;

export type TableData = {
  columnHeadings: TableColumns;
  rowData: TableRow[];
};

export type TopicData = {
  context: {
    heading: string;
    subheading: string;
  };
  tableData: TableData;
  accuracy: string;
};

export type DioceseData = {
  massAttendance: TopicData;
  conversions: TopicData;
};

export type DioceseDb = {
  "Arundel & Brighton": DioceseData;
  Birmingham: DioceseData;
  Brentwood: DioceseData;
  "Cardiff-Menevia": DioceseData;
  Clifton: DioceseData;
  "East Anglia": DioceseData;
  Hallam: DioceseData;
  "Hexham & Newcastle": DioceseData;
  Lancaster: DioceseData;
  Leeds: DioceseData;
  Liverpool: DioceseData;
  Middlesbrough: DioceseData;
  Northampton: DioceseData;
  Nottingham: DioceseData;
  Plymouth: DioceseData;
  Portsmouth: DioceseData;
  Salford: DioceseData;
  Shrewsbury: DioceseData;
  Southwark: DioceseData;
  Westminster: DioceseData;
  Wrexham: DioceseData;
   // [key in DioceseName]: DioceseData; //TODO: Implement for every diocese
};
