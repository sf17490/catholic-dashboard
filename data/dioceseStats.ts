import { DioceseDb, DioceseDbKey } from "./dataTypes";
import { southwarkData } from "./dioceses/southwark";
import { westminsterData } from "./dioceses/westminster";

export const DioceseSimpleDb: DioceseDb = {
  //"Arundel & Brighton" : southwarkData,
  Southwark: southwarkData,
  Westminster: westminsterData,
};

export function getDioceseData(keyName: DioceseDbKey) {
  return DioceseSimpleDb[keyName];
}
