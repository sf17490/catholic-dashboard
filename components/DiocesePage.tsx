import React from "react";
import TopicSection from "@/components/TopicSection";
import { getDioceseData } from "@/data/dioceseStats";
import { DioceseDbKey } from "@/data/dataTypes";

export type DiocesePageProps = {
  diocese: DioceseDbKey;
};

function DiocesePage({ diocese }: DiocesePageProps) {
  const dioceseData = getDioceseData(diocese);
  return (
    <div className="m-10">
      <h1 data-testid={`${diocese}PageTitle`} className="govuk-heading-xl">
        Diocese of {diocese}
      </h1>
      <div data-testid="citation">
        <b>Citation</b>
        <p data-testid="citationText">
          The data on this page comes from{" "}
          <i>
            <a
              data-testid="citationLink"
              className="govuk-link"
              href="https://www.crs.org.uk/catholicism-in-numbers"
            >
              Catholicism in Numbers
            </a>
          </i>
          , a project by the Catholic Record Society.
        </p>
      </div>
      <TopicSection
        rootTestId={`${diocese}Conversions`}
        topicName="Conversions"
        topicData={dioceseData.conversions}
      />
      <TopicSection
        rootTestId={`${diocese}MassAttendance`}
        topicName="Mass Attendance"
        topicData={dioceseData.massAttendance}
      />
    </div>
  );
}

export default DiocesePage;
