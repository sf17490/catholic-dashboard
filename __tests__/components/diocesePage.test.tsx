import DiocesePage from "@/components/DiocesePage";
import { DioceseDbKey, validDioceseNames } from "@/data/dataTypes";
import { DioceseSimpleDb } from "@/data/dioceseStats";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

describe("Diocese page", () => {
  it("includes a page Title", () => {
    render(<DiocesePage diocese="Southwark" />);
    const pageTitle = screen.getByTestId("SouthwarkPageTitle");
    expect(pageTitle).toBeInTheDocument();
    expect(pageTitle).toHaveTextContent("Diocese of Southwark");
  });

  it("includes a citation", () => {
    render(<DiocesePage diocese="Southwark" />);
    const citation = screen.getByTestId("citationText");
    expect(citation).toBeInTheDocument();
  });

  const dioceses: DioceseDbKey[] = [
    "Arundel & Brighton",
    "Birmingham",
    "Brentwood",
    "Clifton",
    "Cardiff-Menevia",
    "East Anglia",
    "Hallam",
    "Hexham & Newcastle",
    "Lancaster",
    "Leeds",
    "Liverpool",
    "Middlesbrough",
    "Northampton",
    "Southwark",
    "Westminster",
  ];
  
  dioceses.forEach((diocese) => {
    it(`has the correct data & formatting for the diocese of ${diocese}`, () => {
      const { asFragment } = render(<DiocesePage diocese={diocese} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it("explains that Cardiff and Menevia were merged in 2024", () => {
    render(<DiocesePage diocese="Cardiff-Menevia" />);

    expect(screen.getByTestId("dioceseNoteHeader")).toBeInTheDocument();

    const dioceseNote = screen.getByTestId("dioceseNote");
    expect(dioceseNote).toHaveTextContent(
      "The Diocese of Cardiff-Menevia was formed on 12 September 2024 when Pope Francis merged the Archdiocese of Cardiff with the Diocese of Menevia. Data prior to this date has been calculated by adding the statistics from Cardiff to the statistics from Menevia. Separate data for each diocese can be found in the Catholicism In Numbers project by the Catholic Record Society."
    );
  });

  it("does not talk about cardiff-Menevia on the other diocese Pages", () => {
    render(<DiocesePage diocese="Birmingham" />);

    expect(screen.getByTestId("maybeDioceseNote")).toHaveTextContent("")
  });
});
