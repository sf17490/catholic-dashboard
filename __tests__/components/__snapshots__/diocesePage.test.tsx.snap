// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diocese page has the correct data & formatting for the diocese of Southwark 1`] = `
<DocumentFragment>
  <div
    class="m-10"
  >
    <h1
      class="govuk-heading-xl"
      data-testid="SouthwarkPageTitle"
    >
      Diocese of Southwark
    </h1>
    <div
      data-testid="citation"
    >
      <b>
        Citation
      </b>
      <p
        data-testid="citationText"
      >
        The data on this page comes from 
        <i>
          <a
            class="govuk-link"
            data-testid="citationLink"
            href="https://www.crs.org.uk/catholicism-in-numbers"
          >
            Catholicism in Numbers
          </a>
        </i>
        , a project by the Catholic Record Society.
      </p>
    </div>
    <div>
      <h2
        class="govuk-heading-l"
        data-testid="SouthwarkConversionsTopicSectionHeading"
      >
        Conversions
      </h2>
      <strong
        data-testid="SouthwarkConversionsAccuracySubheading"
      >
        Note on accuracy of data
      </strong>
      <p
        data-testid="SouthwarkConversionsAccuracy"
      >
        The data below records receptions into the Catholic Church. It does not include figures for adult baptisms.
      </p>
      <div
        class="bg-[var(--colour-grey-3)]"
        data-testid="SouthwarkConversionsDataDisplayBox"
      >
        <div
          class="m-6 pt-2 pb-0.25"
          data-testid="SouthwarkConversionsDataDisplayBoxContents"
        >
          <div
            data-testid="SouthwarkConversionsChartSection"
          >
            <div>
              <h3
                class="govuk-heading-m mb-1"
                role="contextHeader"
              >
                Conversions to Catholicism in the Diocese of Southwark
              </h3>
              <i
                role="contextParagraph"
              >
                Receptions into the Catholic Church in the Diocese of Southwark
              </i>
              <div
                role="graph"
              >
                <div>
                  <img
                    data-testid="placeholderChartImage"
                    src="images/minimalist_National_Conversions.png"
                    width="50%"
                  />
                  <p
                    data-testid="placeholderText"
                  >
                    Loading graph...
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              data-testid="SouthwarkConversionsTableSection"
            >
              <div>
                <table
                  class="govuk-table"
                  data-testid="table"
                >
                  <thead
                    class="govuk-table__head"
                    data-testid="columnHeadings"
                  >
                    <tr
                      class="govuk-table__row"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="keyColumn"
                        scope="col"
                      >
                        Year
                      </th>
                      <th
                        class="govuk-table__header"
                        data-testid="valueColumn"
                        scope="col"
                      >
                        Converts to Catholicism
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="govuk-table__body"
                    data-testid="rows"
                  >
                    <tr
                      class="govuk-table__row"
                      data-testid="row1"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row1key"
                        scope="row"
                      >
                        2022
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row1Value"
                      >
                        206
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row2"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row2key"
                        scope="row"
                      >
                        2021
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row2Value"
                      >
                        132
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row3"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row3key"
                        scope="row"
                      >
                        2019
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row3Value"
                      >
                        358
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row4"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row4key"
                        scope="row"
                      >
                        2010
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row4Value"
                      >
                        451
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row5"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row5key"
                        scope="row"
                      >
                        2000
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row5Value"
                      >
                        128
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h2
        class="govuk-heading-l"
        data-testid="SouthwarkMassAttendanceTopicSectionHeading"
      >
        Mass Attendance
      </h2>
      <strong
        data-testid="SouthwarkMassAttendanceAccuracySubheading"
      >
        Note on accuracy of data
      </strong>
      <p
        data-testid="SouthwarkMassAttendanceAccuracy"
      >
        The accuracy with which attendance is counted may vary. A person may accidentally be counted twice and figures may overstate or understate where an estimate is required, such as at very large services or if mechanical means of counting fail. Additionally, approaches to counting may vary across churches in a diocese or nation.
      </p>
      <div
        class="bg-[var(--colour-grey-3)]"
        data-testid="SouthwarkMassAttendanceDataDisplayBox"
      >
        <div
          class="m-6 pt-2 pb-0.25"
          data-testid="SouthwarkMassAttendanceDataDisplayBoxContents"
        >
          <div
            data-testid="SouthwarkMassAttendanceChartSection"
          >
            <div>
              <h3
                class="govuk-heading-m mb-1"
                role="contextHeader"
              >
                Typical Sunday Mass Attendance
              </h3>
              <i
                role="contextParagraph"
              >
                Typical number of people attending Sunday Mass in the Diocese of Southwark
              </i>
              <div
                role="graph"
              >
                <div>
                  <img
                    data-testid="placeholderChartImage"
                    src="images/minimalist_National_Mass_Attendance.png"
                    width="50%"
                  />
                  <p
                    data-testid="placeholderText"
                  >
                    Loading graph...
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              data-testid="SouthwarkMassAttendanceTableSection"
            >
              <div>
                <table
                  class="govuk-table"
                  data-testid="table"
                >
                  <thead
                    class="govuk-table__head"
                    data-testid="columnHeadings"
                  >
                    <tr
                      class="govuk-table__row"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="keyColumn"
                        scope="col"
                      >
                        Year
                      </th>
                      <th
                        class="govuk-table__header"
                        data-testid="valueColumn"
                        scope="col"
                      >
                        Number attending Mass
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="govuk-table__body"
                    data-testid="rows"
                  >
                    <tr
                      class="govuk-table__row"
                      data-testid="row1"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row1key"
                        scope="row"
                      >
                        2022
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row1Value"
                      >
                        57360
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row2"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row2key"
                        scope="row"
                      >
                        2021
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row2Value"
                      >
                        48188
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row3"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row3key"
                        scope="row"
                      >
                        2018
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row3Value"
                      >
                        80178
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row4"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row4key"
                        scope="row"
                      >
                        2008
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row4Value"
                      >
                        94217
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row5"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row5key"
                        scope="row"
                      >
                        1999
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row5Value"
                      >
                        94467
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Diocese page has the correct data & formatting for the diocese of Westminster 1`] = `
<DocumentFragment>
  <div
    class="m-10"
  >
    <h1
      class="govuk-heading-xl"
      data-testid="WestminsterPageTitle"
    >
      Diocese of Westminster
    </h1>
    <div
      data-testid="citation"
    >
      <b>
        Citation
      </b>
      <p
        data-testid="citationText"
      >
        The data on this page comes from 
        <i>
          <a
            class="govuk-link"
            data-testid="citationLink"
            href="https://www.crs.org.uk/catholicism-in-numbers"
          >
            Catholicism in Numbers
          </a>
        </i>
        , a project by the Catholic Record Society.
      </p>
    </div>
    <div>
      <h2
        class="govuk-heading-l"
        data-testid="WestminsterConversionsTopicSectionHeading"
      >
        Conversions
      </h2>
      <strong
        data-testid="WestminsterConversionsAccuracySubheading"
      >
        Note on accuracy of data
      </strong>
      <p
        data-testid="WestminsterConversionsAccuracy"
      >
        The data below records receptions into the Catholic Church. It does not include figures for adult baptisms.
      </p>
      <div
        class="bg-[var(--colour-grey-3)]"
        data-testid="WestminsterConversionsDataDisplayBox"
      >
        <div
          class="m-6 pt-2 pb-0.25"
          data-testid="WestminsterConversionsDataDisplayBoxContents"
        >
          <div
            data-testid="WestminsterConversionsChartSection"
          >
            <div>
              <h3
                class="govuk-heading-m mb-1"
                role="contextHeader"
              >
                Conversions to Catholicism in the Diocese of Westminster
              </h3>
              <i
                role="contextParagraph"
              >
                Receptions into the Catholic Church in the Diocese of Westminster
              </i>
              <div
                role="graph"
              >
                <div>
                  <img
                    data-testid="placeholderChartImage"
                    src="images/minimalist_National_Conversions.png"
                    width="50%"
                  />
                  <p
                    data-testid="placeholderText"
                  >
                    Loading graph...
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              data-testid="WestminsterConversionsTableSection"
            >
              <div>
                <table
                  class="govuk-table"
                  data-testid="table"
                >
                  <thead
                    class="govuk-table__head"
                    data-testid="columnHeadings"
                  >
                    <tr
                      class="govuk-table__row"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="keyColumn"
                        scope="col"
                      >
                        Year
                      </th>
                      <th
                        class="govuk-table__header"
                        data-testid="valueColumn"
                        scope="col"
                      >
                        Converts to Catholicism
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="govuk-table__body"
                    data-testid="rows"
                  >
                    <tr
                      class="govuk-table__row"
                      data-testid="row1"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row1key"
                        scope="row"
                      >
                        2022
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row1Value"
                      >
                        227
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row2"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row2key"
                        scope="row"
                      >
                        2021
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row2Value"
                      >
                        193
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row3"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row3key"
                        scope="row"
                      >
                        2019
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row3Value"
                      >
                        298
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row4"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row4key"
                        scope="row"
                      >
                        2010
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row4Value"
                      >
                        628
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row5"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row5key"
                        scope="row"
                      >
                        2000
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row5Value"
                      >
                        541
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h2
        class="govuk-heading-l"
        data-testid="WestminsterMassAttendanceTopicSectionHeading"
      >
        Mass Attendance
      </h2>
      <strong
        data-testid="WestminsterMassAttendanceAccuracySubheading"
      >
        Note on accuracy of data
      </strong>
      <p
        data-testid="WestminsterMassAttendanceAccuracy"
      >
        The accuracy with which attendance is counted may vary. A person may accidentally be counted twice and figures may overstate or understate where an estimate is required, such as at very large services or if mechanical means of counting fail. Additionally, approaches to counting may vary across churches in a diocese or nation.
      </p>
      <div
        class="bg-[var(--colour-grey-3)]"
        data-testid="WestminsterMassAttendanceDataDisplayBox"
      >
        <div
          class="m-6 pt-2 pb-0.25"
          data-testid="WestminsterMassAttendanceDataDisplayBoxContents"
        >
          <div
            data-testid="WestminsterMassAttendanceChartSection"
          >
            <div>
              <h3
                class="govuk-heading-m mb-1"
                role="contextHeader"
              >
                Typical Sunday Mass Attendance
              </h3>
              <i
                role="contextParagraph"
              >
                Typical number of people attending Sunday Mass in the Diocese of Westminster
              </i>
              <div
                role="graph"
              >
                <div>
                  <img
                    data-testid="placeholderChartImage"
                    src="images/minimalist_National_Mass_Attendance.png"
                    width="50%"
                  />
                  <p
                    data-testid="placeholderText"
                  >
                    Loading graph...
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              data-testid="WestminsterMassAttendanceTableSection"
            >
              <div>
                <table
                  class="govuk-table"
                  data-testid="table"
                >
                  <thead
                    class="govuk-table__head"
                    data-testid="columnHeadings"
                  >
                    <tr
                      class="govuk-table__row"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="keyColumn"
                        scope="col"
                      >
                        Year
                      </th>
                      <th
                        class="govuk-table__header"
                        data-testid="valueColumn"
                        scope="col"
                      >
                        Number attending Mass
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="govuk-table__body"
                    data-testid="rows"
                  >
                    <tr
                      class="govuk-table__row"
                      data-testid="row1"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row1key"
                        scope="row"
                      >
                        2022
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row1Value"
                      >
                        90202
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row2"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row2key"
                        scope="row"
                      >
                        2021
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row2Value"
                      >
                        74308
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row3"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row3key"
                        scope="row"
                      >
                        2018
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row3Value"
                      >
                        127340
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row4"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row4key"
                        scope="row"
                      >
                        2008
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row4Value"
                      >
                        151668
                      </td>
                    </tr>
                    <tr
                      class="govuk-table__row"
                      data-testid="row5"
                    >
                      <th
                        class="govuk-table__header"
                        data-testid="row5key"
                        scope="row"
                      >
                        1999
                      </th>
                      <td
                        class="govuk-table__cell"
                        data-testid="row5Value"
                      >
                        154042
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
